### PR TITLE
Checking things off

### PR DESCRIPTION
**Pull request by jamesaimonetti**
check head of list if passing in sub-fields

Use Token-Info

pass endpoint data into getter

tickie

refactor to find realm

dialzyer

### CODE REVIEW

File: kazoo_services/src/kz_services.erl [hunk 1]
```diff
@@ -912,7 +912,7 @@ create(AccountId) ->
           kz_term:api_object().
 create(_AccountId, {'error', 'not_found'}) ->
     lager:info("failed to find account database for ~s", [_AccountId]),
-    'undefined';
+    {'error', 'not_found'};
```
Comment (lazedo) [inline kazoo_services/src/kz_services.erl]: this change looks wrong according to the `spec`

**Review by icehess — APPROVED** [review]

**Comment by lazedo** [comment]
There was still in progress for this

**Comment by icehess** [comment]
@lazedo Sorry I did'nt see anything blocking this. What is still in progress?

File: kazoo_amqp/src/api/kapi_registration.erl [hunk 2]
```diff
@@ -179,12 +179,17 @@ query_req_definition() ->
...
-                                        lists:all(fun(F) -> lists:member(F, Allowed) end, Fs);
+                                        Allowed = [<<"Bridge-RURI">>
+                                                  | kapi_definition:optional_headers(SuccessDef)
+                                                   ++ kapi_definition:required_headers(SuccessDef)
+                                                  ],
+                                        lists:all(fun(<<Field/binary>>) -> lists:member(Field, Allowed);
+                                                     ([<<Field/binary>> | _]) -> lists:member(Field, Allowed)
+                                                  end
+                                                 ,Fields
+                                                 );
```
Comment (lazedo) [inline kazoo_amqp/src/api/kapi_registration.erl:192]: @jamesaimonetti can you please revisit this, handling [[<<"field">>]] doesn't look right

**Pull request merged**